commons/src/main/java/com/powsybl/commons/datasource/DataSourceUtil.java

```diff
@@ -62,20 +63,41 @@ static DataSource createDataSource(Path directory, String basename, String dataE
         return dataSourceBuilder.build();
     }
 
-    static DataSource createDataSource(Path directory, String fileNameOrBaseName, DataSourceObserver observer) {
-        Objects.requireNonNull(directory);
-        Objects.requireNonNull(fileNameOrBaseName);
+    static DataSource createDataSource(Path fileOrDirectory, DataSourceObserver observer) {
+        Objects.requireNonNull(fileOrDirectory);
+
+        DataSourceBuilder dataSourceBuilder = new DataSourceBuilder().withObserver(observer);
+        if (Files.isDirectory(fileOrDirectory)) {
+            dataSourceBuilder.withDirectory(fileOrDirectory)
+                .withBaseName(fileOrDirectory.getFileName().toString())
```

Comment:
```suggestion
// we use the full directoryname here instead of a parsed representation using FileInformation
// this is required for now to allow directories with dots but without the dataextension (like "microgrid.v1.complete/")
// otherwise Fileinformation interprets the part after the dot as the dataextension. This has the drawback that
// directories with a dataextension (e.g. "network.xiidm/") will be rejected/mishandled because we will use
// a file named "network.xiidm.xiidm" inside it
// NOTE: this does not occur if the directory has a dot and also has the dataextension: "microgrid.v1.complete.xml/" works
// NOTE2: Maybe this can be fixed in the future by looking more at what is inside the directory and we can remove this
                .withBaseName(fileOrDirectory.getFileName().toString())
```

Reply:
This should go in the javadoc more than in a simple comment